cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -50,6 +50,7 @@ namespace bp = boost::process;
 namespace gc = google::cloud;
```

Comment:
Can you add a test for FileSystemFromUri somewhere in this file?

Reply:
added a test to make sure it can instantiate the file system.  there are detailed test for the FromUri implementations on GCS already.